### DISCUSSION TITLE
API - get all assets

### DISCUSSION THREAD

**Discussion by wardwygaerts** [immich-app/immich#10888] [Q&A] [answered]
I have some scripts to create albums and populate them, but I noticed the API changed.
To grab all assets I used GET `/api/asset`, but this doesn't work anymore. In the docs I saw `asset` changed to `assets`, but still receiving this error:
```
{
  "message": "Cannot GET /api/assets",
  "error": "Not Found",
  "statusCode": 404,
  "correlationId": "9y1fov1g"
}
```

other API calls are still working eg `/api/assets/467bb8c1-5fae-4a9e-a7a9-6c8cb0a74782` gives me all info of this asset.

**Comment by bo0tzz** [comment, marked as answer]
You'll want to use the searchMetadata endpoint instead

**Reply by wardwygaerts** [reply]
Thanks! I had to build some loops, because it outputs assets per 250

**Reply by bo0tzz** [reply]
You can set the amount of returned assets with the `size` param. But yes - you should be looping through and fetching assets a bit at a time.

**Reply by rehack** [reply]
> You'll want to use the searchMetadata endpoint instead

Thank you very much！ I can retrieve all resources through the Axios searchMetadata endpoint on the web, and the resulting JSON like:
``` json
{
    "id": "fa9996c5-2915-48e6-aa75-83511a358297",
    "deviceAssetId": "1693934",
    "ownerId": "708ed3e2-90fe-4163-874f-0be1351d903f",
    "deviceId": "91846063e500f77d667cc9b443a03d58u78de1c768a4fc189d8c97f1ea619597",
    "libraryId": null,
    "type": "IMAGE",
    "originalPath": "upload/library/admin/2024/09/IMG_20240904_173433_1.jpg",
    "originalFileName": "IMG_20240904_173433_1.jpg",
    "originalMimeType": "image/jpeg",
    "thumbhash": "EQgKDYCcqphrd2dveIeYfaxI/5jV",
    "fileCreatedAt": "2024-09-04T17:34:36.804Z",
    "fileModifiedAt": "2024-09-04T09:34:36.000Z",
    "localDateTime": "2024-09-04T17:34:36.804Z",
    "updatedAt": "2024-09-19T04:23:30.785Z",
    "isFavorite": false,
    "isArchived": false,
    "isTrashed": false,
    "duration": "0:00:00.00000",
    "livePhotoVideoId": null,
    "people": [],
    "checksum": "Vldk/GPUFZgkHdgaXYhJViJD+21=",
    "isOffline": false,
    "hasMetadata": true,
    "duplicateId": null,
    "resized": true
}
```

but how to enumerate all assets, Render and display all image resources on a web page?

**Comment by radityaK** [comment]
Hi, i this API been removed `/api/assets` ? I can't find it on docs, also I wonder which API I could use to fetch all list of asset.

**Reply by bo0tzz** [reply]
You can use the search endpoints, if you don't apply any filters they'll return all the assets.

**Reply by radityaK** [reply]
which one? 

![image](https://github.com/user-attachments/assets/f34d5b14-3409-431f-8a5f-2c0e8bbf49e0)

**Reply by rehack** [reply]
> 如果

I use immich as a powerful image repository，it's really great.

I want to display the images on the immich server on my own website by calling the immich API

I tried to call searchMetadata endpoint in my web page, and successfully returned JSON data:
```json
{

"id": "fa9996c5-2915-48e6-aa75-83511a358297",

"deviceAssetId": "1693934",

"ownerId": "708ed3e2-90fe-4163-874f-0be1351d903f",

"deviceId": "91846063e500f77d667cc9b443a03d58u78de1c768a4fc189d8c97f1ea619597",

"libraryId": null,

"type": "IMAGE",

"originalPath": "upload/library/admin/2024/09/IMG\_20240904\_173433\_1.jpg",

"originalFileName": "IMG\_20240904\_173433\_1.jpg",

"originalMimeType": "image/jpeg",

"thumbhash": "EQgKDYCcqphrd2dveIeYfaxI/5jV",

"fileCreatedAt": "2024-09-04T17:34:36.804Z",

"fileModifiedAt": "2024-09-04T09:34:36.000Z",

"localDateTime": "2024-09-04T17:34:36.804Z",

"updatedAt": "2024-09-19T04:23:30.785Z",

"isFavorite": false,

"isArchived": false,

"isTrashed": false,

"duration": "0:00:00.00000",

"livePhotoVideoId": null,

"people": \[\],

"checksum": "Vldk/GPUFZgkHdgaXYhJViJD+21=",

"isOffline": false,

"hasMetadata": true,

"duplicateId": null,

"resized": true

}
```
But when I enumerate these image resources on the webpage, the images will not be displayed properly. Why? What did I miss?

**Reply by radityaK** [reply]
hit this endpoint `https://immich.app/docs/api/view-asset` then you need to add request header when you want to display the image.

**Reply by radityaK** [reply]
@bo0tzz anyway, why my access token frequently invalid? I got the accessToken after hitting login api. And what is the difference between token from `createKey` api?
